src/brotli.rs

```diff
@@ -27,48 +29,40 @@ impl<S: Stream<Item = Result<Bytes>>> Stream for CompressedStream<S> {
     fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Result<Bytes>>> {
         const OUTPUT_BUFFER_SIZE: usize = 8_000;
 
-        let (inner, flushing, input_buffer, compress) = unsafe {
-            let CompressedStream {
-                inner,
-                flushing,
-                input_buffer,
-                compress,
-            } = self.get_unchecked_mut();
-            (Pin::new_unchecked(inner), flushing, input_buffer, compress)
-        };
+        let this = self.project();
 
-        if input_buffer.is_empty() {
-            if *flushing {
-                return Poll::Ready(None);
-            } else if let Some(bytes) = ready!(inner.poll_next(cx)) {
-                *input_buffer = bytes?;
-            } else {
-                *flushing = true;
-            }
+        if *this.flushing {
+            return Poll::Ready(None);
         }
 
+        let input_buffer = if let Some(bytes) = ready!(this.inner.poll_next(cx)) {
+            bytes?
+        } else {
+            *this.flushing = true;
+            Bytes::new()
+        };
+
         let mut compressed_output = BytesMut::with_capacity(OUTPUT_BUFFER_SIZE);
         let input_ref = &mut input_buffer.as_ref();
         let output_ref = &mut &mut [][..];
         loop {
-            let status = compress.compress(
-                if *flushing {
+            let status = this.compress.compress(
+                if *this.flushing {
                     CompressOp::Finish
                 } else {
                     CompressOp::Process
                 },
                 input_ref,
                 output_ref,
             )?;
-            while let Some(buf) = dbg!(compress.take_output(None)) {
+            while let Some(buf) = dbg!(this.compress.take_output(None)) {
```

Comment:
I did forget to leave this out -- You can remove the surrounding dbg! macro.

Reply:
Oh, heh, I’ll remove it in a later PR